csharp/src/Ice/Communicator.cs

```diff
@@ -2017,8 +2018,7 @@ internal void SetServerProcessProxy(ObjectAdapter adminAdapter, Identity adminId
                         s.Append("the server is not known to the locator registry");
                         Logger.Trace(TraceLevels.LocationCat, s.ToString());
                     }
-
-                    throw new InitializationException("Locator knows nothing about server `" + serverId + "'");
+                    throw;
```

Comment:
This should be merged with the catch block below if we no longer want to notify the user about the invalid configuration.